### DISCUSSION TITLE
Possible to set initial robot positions and orientations?

### DISCUSSION THREAD

**Discussion by dEgolf** [robocode-dev/tank-royale#3] [Q&A] [answered]
I've been trying to create a very simple robot, and have been running in to some issues. For consistency in debugging (to reproduce problems), I was wondering: is it possible to set the initial position and orientation of robots? Thanks!

**Comment by flemming-n-larsen** [comment, marked as answer]
Currently, this is not possible. But this feature is available with the original version of Robocode, and adding it to Tank Royale is a great idea. So I will work on getting this feature into Robocode as well. 🙂

**Comment by flemming-n-larsen** [comment]
I have just release [version 0.11.0](https://github.com/robocode-dev/tank-royale/releases/tag/v0.11.0) which is exactly about implementing this feature:

- Implemented initial position feature:
    - Add an `initialPosition` field in the JSON file for your bot with a value like `50,50,90` to request a starting
      coordinate at (50,50) with the bot heading toward 90°.
    - A `--enable-initial-position` (or the short version `-I`) must be set to enable the initial positions feature on
      the server. If initial position is not enabled, the bot will start at a random position.
    - Initial positions can be set from the menu with Config → Debug Options → Enable initial start position.
    - The Target.json file of the Target sample has been updated with an `initialPosition` to demonstrate this feature.
    
 Enjoy! 🙂

**Reply by dEgolf** [reply]
Wonderful! This will be very useful for future debugging!
